ansible_collections/arista/avd/roles/eos_designs/templates/base/static-routes.j2

```diff
@@ -1,12 +1,14 @@
 static_routes:
-{% if mgmt_destination_networks is arista.avd.defined and mgmt_destination_networks | length > 0 %}
-{%     for mgmt_destination_network in mgmt_destination_networks %}
+{% if mgmt_gateway is arista.avd.defined %}
```

Comment:
Just to verify, would this be the correct behavior in these cases:
* ip + gateway == route
* ip only = no route (because we don't know the next hop?)
* gateway only = ? (right now we have `gateway only = route` but then if there is no IP it may not be very useful indeed)
* neither == no route

Regarding the MGMT vrf, it becomes sneaky because we have in some places like name-servers (and maybe other I would need to analyze further) the mgmt_vrf expected to be defined and because we are giving a default value this is always generated:
https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_designs/templates/base/vrfs.j2#L3

Example where the VRF name will be used even if we don't render it:
* https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_designs/templates/base/name-servers.j2#L5

Reply:
Agree on your behavior descriptions.
You are right on the other templates. Would it stop the config of MGMT vrf if I set `mgmt_interface_vrf: null`?

Reply:
`mgmt_interface_vrf: null` raises a not very explicit error:

```
TASK [arista.avd.eos_designs : Generate device configuration in structured format] ***
An exception occurred during task execution. To see the full traceback, use -vvv. The error was:       ^
fatal: [no_mgmt_interface -> localhost]: FAILED! => {"msg": "Unexpected failure during module execution.", "stdout": ""}
```

Reply:
Let's keep this PR only looking at mgmt_gateway and punt the issue of mgmt_interface_vrf to another PR.

Reply:
So i had started on something in here: https://github.com/gmuloc/ansible-avd/tree/mgmt-interface-vrf building upon 5b1eef5 but this was not completed yet we can postpone this to a new issue / PR indeed